version: '3.8'

services:
  web:
    build: .
    volumes:
      - .:/app
    ports:
      - "8006:8006"
    environment:
      - DJANGO_DEBUG=False
      - ALLOWED_HOSTS=192.168.15.215,localhost,127.0.0.1,0.0.0.0
      - MYSQL_DATABASE=alkosto_verify_db
      - MYSQL_USER=alkosto_user
      - MYSQL_PASSWORD=alkosto_password
      - MYSQL_HOST=db
      - MYSQL_PORT=3306
      - DJANGO_SETTINGS_MODULE=alkosto_verify.settings
      - DOCKER_CONTAINER=True
    depends_on:
      - db
    restart: on-failure

  db:
    image: mysql:8.0
    volumes:
      - mysql_data:/var/lib/mysql
    environment:
      - MYSQL_ROOT_PASSWORD=root_password
      - MYSQL_DATABASE=alkosto_verify_db
      - MYSQL_USER=alkosto_user
      - MYSQL_PASSWORD=alkosto_password
      - MYSQL_ROOT_HOST=%
    ports:
      - "3307:3306"
    command: --default-authentication-plugin=mysql_native_password
    healthcheck:
      test: ["CMD", "mysqladmin", "ping", "-h", "localhost", "-u", "alkosto_user", "-palkosto_password"]
      interval: 10s
      timeout: 5s
      retries: 5

volumes:
  mysql_data: